### PR TITLE
chore: add unit tests and remove unused variables

### PR DESCRIPTION
**Pull request by wifimug**

### CODE REVIEW

**Comment by DanielSwarup** [comment]
The cache binary files should be removed and not pushed to Git

File: tests/controllers/test_camera_controller.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import unittest
+import bpy
+from src.controllers.camera_controller import CameraController
+
+
+class BlenderScriptTest(unittest.TestCase):
```
Comment (DanielSwarup) [inline tests/controllers/test_camera_controller.py:6]: This should be named "CamerControllerTest" or something more relevant

File: tests/controllers/test_camera_controller.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    def test_setup_camera_collection_name(self):
+        # Create camera controller instance
+        collection = self.camera_controller.setup_camera(self.camera_name, self.camera_location, self.camera_rotation, self.collection)
+        # Checks that the collection is named correctly
+        self.assertTrue(collection.name == self.collection)
+
+    def test_setup_camera_collection_scene(self):
+        for collection in bpy.data.collections:
+            if collection.name == self.collection:
+                # Checks that the colelction exists in the scene
```
Comment (DanielSwarup) [inline tests/controllers/test_camera_controller.py:25]: A test case should be added to check the rotation and location of the camera

File: tests/controllers/test_crop_controller.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
+import unittest
+import yaml
+import os
+from src.controllers.crop_controller import CropController
+from src.controllers.yaml_reader import YamlReader
+
+
+class BlenderScriptTest(unittest.TestCase):
```
Comment (DanielSwarup) [inline tests/controllers/test_crop_controller.py:8]: More relevant test name required

File: tests/controllers/test_crop_controller.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+            yaml.safe_dump(test_data, file)
+
+
+        input_data = YamlReader().read_file(self.test_file)
+
+        crop_controller = CropController(input_data["crop"], self.collection)
+        material, segmentation_id = crop_controller.assign_crop_type("red")
+        self.assertEquals(material.name, self.expected_material_name)
+
+    def test_setup_crops_material_seg_id(self):
```
Comment (DanielSwarup) [inline tests/controllers/test_crop_controller.py:44]: (Optional) A test for ensuring the number of crops rendered is correct

File: tests/test_launch.py [hunk 5]
```diff
@@ -13,11 +13,11 @@ def setUp(self):
         self.test_output = "tests/expected_output.png"
         self.expected_output_file = os.getcwd() + "/" + self.test_output
 
-    def tearDown(self):
-        # Clean up test environment
-        os.remove(self.test_file)
-        os.remove(self.expected_output_file)
-        os.remove("tests/expected_output_seg.png")
+    # def tearDown(self):
```
Comment (DanielSwarup) [inline tests/test_launch.py:16]: This needs to be commented out so files are cleaned

**Review by DanielSwarup — CHANGES_REQUESTED** [review]

File: tests/controllers/test_camera_controller.py [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+        self.camera_controller.setup_camera(self.camera_name, self.camera_location, self.camera_rotation, self.collection)
+        camera_loc = bpy.data.objects[self.camera_name].location
+        x, y, z = camera_loc
+        self.assertEqual(self.camera_location, (x, y, z))
+
+    def test_setup_camera_collection_rotation(self):
+        self.camera_controller.setup_camera(self.camera_name, self.camera_location, self.camera_rotation, self.collection)
+        camera_rot = bpy.data.objects[self.camera_name].rotation_euler
+        # only getting first 3 values - they are relevant to rotation, after that it is the perspective
+        self.assertEqual(self.camera_rotation, camera_rot[0:3])
```
Comment (DanielSwarup) [inline tests/controllers/test_camera_controller.py:41]: Great!

File: tests/controllers/test_crop_controller.py [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+        # Set up test environment
+        self.test_file = "tests/test_data.yml"
+        self.test_output = "tests/test_output.png"
+        self.collection = "Test Collection"
+        self.expected_material_name = "Red"
+        self.expected_segmentation_id = 1
+
+    def tearDown(self):
+        # Clean up test environment
+        os.remove(self.test_file)
```
Comment (DanielSwarup) [inline tests/controllers/test_crop_controller.py:20]: Missing teardown of "tests/test_output.png" i.e. self.test_output

File: tests/controllers/test_camera_controller.py [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+    def test_setup_camera_collection_scene(self):
+        collection = self.camera_controller.setup_camera(self.camera_name, self.camera_location, self.camera_rotation, self.collection)
+        for collection in bpy.data.collections:
+            if collection.name == self.collection:
+                # Checks that the colelction exists in the scene
+                self.assertTrue(collection.name == self.collection)
+
+    def test_setup_camera_collection_location(self):
+        self.camera_controller.setup_camera(self.camera_name, self.camera_location, self.camera_rotation, self.collection)
+        camera_loc = bpy.data.objects[self.camera_name].location
```
Comment (DanielSwarup) [inline tests/controllers/test_camera_controller.py:33]: These look good!

**Review by DanielSwarup — CHANGES_REQUESTED** [review]

File: tests/controllers/test_crop_controller.py (hunk 7 above)
Comment (wifimug) [reply tests/controllers/test_crop_controller.py:20]: There is no teardown for "test/test_output.png" as it is not created while checking. I will delete the "tests/test_output.png" in the setup

Comment (DanielSwarup) [reply tests/controllers/test_crop_controller.py:20]: Thank you for the clarification

**Review by DanielSwarup — APPROVED** [review]
LGTM!

**Comment by DanielSwarup** [comment]
Unit test is failing!

**Review by DanielSwarup — APPROVED** [review]

**Review by SeanChao — APPROVED** [review]
LGTM!

**Pull request merged**